aspnetcore/blazor/fundamentals/routing.md

```diff
@@ -704,6 +704,35 @@ The following component:
 
 For more information on component disposal, see <xref:blazor/components/component-disposal>.
 
+:::moniker range=">= aspnetcore-9.0"
+
+## Navigation Manager redirect behavior during static server-side rendering (static SSR)
+
+:::moniker-end
+
+:::moniker range=">= aspnetcore-10.0"
+
+In .NET 9 for a redirect during static server-side rendering (static SSR), <xref:Microsoft.AspNetCore.Components.NavigationManager> relied on throwing an exception that was captured by the framework, which converted the error into a redirect. Code that existed after the call to <xref:Microsoft.AspNetCore.Components.NavigationManager.NavigateTo%2A> abruptly stopped, which also didn't match the behavior during interactive rendering. When using Visual Studio, the debugger stopped on the exception, which required deselecting the checkbox for **Break when this exception type is user-handled** in the Visual Studio UI to avoid the debugger stopping for future redirects.
```

Comment:
We ended up making the new static SSR navigation behavior opt-in for .NET 10. The default is still to throw an exception. Setting the `<BlazorDisableThrowNavigationException>true</BlazorDisableThrowNavigationException>` property will switch on the new behavior to no longer throw an exception. This behavior is enabled in the .NET 10 version of the Blazor Web App project template.